src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/LoggerAwarePass.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Psr\Log\LoggerAwareInterface;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Automatically add setLogger method call to any service that implements Psr\Log\LoggerAwareInterface.
+ *
+ * @see http://www.php-fig.org/psr/psr-3/
+ *
+ * @author Gary PEGEOT <garypegeot@gmail.com>
+ */
+class LoggerAwarePass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('logger') && !$container->hasAlias('logger')) {
+            return;
+        }
+
+        $reference = new Reference('logger');
+
+        foreach ($container->getDefinitions() as $definition) {
+            if (!class_exists($definition->getClass()) || !$definition->isAutoconfigured()) {
+                continue;
+            }
+
+            $refl = new \ReflectionClass($definition->getClass());
```

Comment:
should use `$container->getReflectionClass()` and be done earlier, replacing the `class_exists` check by `!$refl` and probably throwing in case the check passes. See e.g. https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Console/DependencyInjection/AddConsoleCommandPass.php#L55